pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -81,65 +85,70 @@ type MasterConfig struct {
 	Informers shared.InformerFactory
 }
 
-// BuildDefaultAPIServer constructs the appropriate APIServer and StorageFactory for the kubernetes server.
+// BuildKubeApiserverOptions constructs the appropriate kube-apiserver run options.
 // It returns an error if no KubernetesMasterConfig was defined.
-func BuildDefaultAPIServer(options configapi.MasterConfig) (*apiserveroptions.ServerRunOptions, genericapiserver.StorageFactory, error) {
-	if options.KubernetesMasterConfig == nil {
-		return nil, nil, fmt.Errorf("no kubernetesMasterConfig defined, unable to load settings")
+func BuildKubeApiserverOptions(masterConfig configapi.MasterConfig) (*kapiserveroptions.ServerRunOptions, error) {
+	if masterConfig.KubernetesMasterConfig == nil {
+		return nil, fmt.Errorf("no kubernetesMasterConfig defined, unable to load settings")
 	}
-	_, portString, err := net.SplitHostPort(options.ServingInfo.BindAddress)
+	_, portString, err := net.SplitHostPort(masterConfig.ServingInfo.BindAddress)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	port, err := strconv.Atoi(portString)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	portRange, err := knet.ParsePortRange(options.KubernetesMasterConfig.ServicesNodePortRange)
+	portRange, err := knet.ParsePortRange(masterConfig.KubernetesMasterConfig.ServicesNodePortRange)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	// Defaults are tested in TestAPIServerDefaults
-	server := apiserveroptions.NewServerRunOptions()
+	server := kapiserveroptions.NewServerRunOptions()
 
 	// Adjust defaults
 	server.EventTTL = 2 * time.Hour
-	server.GenericServerRunOptions.AnonymousAuth = true
-	server.GenericServerRunOptions.ServiceClusterIPRange = net.IPNet(flagtypes.DefaultIPNet(options.KubernetesMasterConfig.ServicesSubnet))
-	server.GenericServerRunOptions.ServiceNodePortRange = *portRange
-	server.GenericServerRunOptions.EnableProfiling = true
-	server.GenericServerRunOptions.MasterCount = options.KubernetesMasterConfig.MasterCount
-
-	server.GenericServerRunOptions.SecurePort = port
-	server.GenericServerRunOptions.InsecurePort = 0
-	server.GenericServerRunOptions.TLSCertFile = options.ServingInfo.ServerCert.CertFile
-	server.GenericServerRunOptions.TLSPrivateKeyFile = options.ServingInfo.ServerCert.KeyFile
-	server.GenericServerRunOptions.ClientCAFile = options.ServingInfo.ClientCA
-
-	server.GenericServerRunOptions.MaxRequestsInFlight = options.ServingInfo.MaxRequestsInFlight
-	server.GenericServerRunOptions.MinRequestTimeout = options.ServingInfo.RequestTimeoutSeconds
-	for _, nc := range options.ServingInfo.NamedCertificates {
-		sniCert := config.NamedCertKey{
+	server.Authentication.WithAnyonymous()
```

Comment:
I put this in here to match what we had before, but we technically don't need it because `kapiserveroptions.NewServerRunOptions()` enables it by default. There's no harm leaving it in, in case kube changes things upstream in the future, though.